modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -135,7 +135,13 @@ const EditorUpload = function (editor: Editor): EditorUpload {
             blobCache.removeByUri(image.src);
             replaceImageUriInView(image, uploadInfo.url);
           } else if (uploadInfo.error) {
-            ErrorReporter.uploadError(editor, uploadInfo.error);
+            if (uploadInfo.error.options.remove) {
+              editor.undoManager.transact(() => {
```

Comment:
I'm not quite sure this is the right place to be doing the transaction, as you'd normally want to do it for the entire upload operation, not per uploaded file.

Reply:
I wonder if what we need is to add a new undo level at the end of the process instead?

Reply:
Let us debate this on the new PR to replace this one.